src/test/java/com/ait/lienzo/client/core/util/GeometryTest.java

```diff
@@ -0,0 +1,206 @@
+        double[] result2 = Geometry.quadraticRoots(-15.4323123d, 1.234412d, 7.354);
+        assertEquals(-1d, result2[0], 0);
+        assertEquals(0.7314654055347382d, result2[1], 0);
+        assertEquals(-1d, result2[2], 0);
+        double[] result3 = Geometry.quadraticRoots(0.00001d, 0.00001d, 0.00001d);
+        assertEquals(-0.5d, result3[0], 0);
+        assertEquals(-0.5d, result3[1], 0);
+        assertEquals(-1d, result3[2], 0);
+    }
+
+    @Test
+    public void testQuadraticToCubic() {
+        double[] result = Geometry.quadraticToCubic(1.2d, 5.3d, 34.3d, 56.5d, 34.2, 45.221d).toArray();
+        assertEquals(1.2d, result[0], 0);
+        assertEquals(5.3d, result[1], 0);
+        assertEquals(23.266666666666662d, result[2], 0);
+        assertEquals(39.43333333333333d, result[3], 0);
+        assertEquals(34.266666666666666d, result[4], 0);
+        assertEquals(52.74033333333333d, result[5], 0);
+        assertEquals(34.2d, result[6], 0);
+        assertEquals(45.221d, result[7], 0);
+    }
+
+    @Test
+    public void testGetBoundingBoxForQuadraticCurve() {
+        BoundingBox result = Geometry.getBoundingBoxForQuadraticCurve(new Point2DArray(new Point2D(1.13d, 3.454d),
+                                                                                       new Point2D(234.4d, 2.2d),
+                                                                                       new Point2D(0.34d, 111d)));
+        assertEquals(0d, result.getMinX(), 0);
+        assertEquals(0d, result.getMinY(), 0);
+        assertEquals(69.66118680175845d, result.getMaxX(), 0);
+        assertEquals(3.454d, result.getMaxY(), 0);
+    }
+
+    @Test
+    public void testGetBoundingBox() {
+        BoundingBox result = Geometry.getBoundingBox(new QuadraticCurve(1.3d, 2.4d, 6.54d, 67.5d));
+        assertEquals(0d, result.getMinX(), 0);
+        assertEquals(0d, result.getMinY(), 0);
+        assertEquals(0.38518518518518524d, result.getMaxX(), 0);
+        assertEquals(0.7111111111111111d, result.getMaxY(), 0);
+    }
+
+    @Test
+    public void testCubicRoots() {
+        double[] result = Geometry.cubicRoots(new double[]{1.2d, 5.3d, 34.3d, 56.5d});
+        assertEquals(0d, result[0], 0);
+        assertEquals(-1d, result[1], 0);
+        assertEquals(-1d, result[2], 0);
+        double[] result1 = Geometry.cubicRoots(new double[]{1.2d, 1.3d, 0d, 0d});
+        assertEquals(0.6388888888888888d, result1[0], 0);
+        assertEquals(0.6388888888888888d, result1[1], 0);
+        assertEquals(-1d, result1[2], 0);
+    }
+
+    @Test
+    public void testNonIntersectLineCurve() {
+        final double[] xval = new double[]{1.2d, 1.2d, 0d, 0d};
+        final double[] yval = new double[]{0d, 0d, 2.1d, 2.1d};
+        final double[] lx = new double[]{0d, 0d};
+        final double[] ly = new double[]{0d, 0d};
+        Point2DArray result = Geometry.intersectLineCurve(xval, yval, lx, ly);
+        assertTrue(result.isEmpty());
```

Comment:
This test looks to easy, it is testing only not intersecting lines. It will be good to add case when two lines will intersect. Is it possible?

Reply:
good point - done! :+1: